blueprints/component/index.js

```diff
@@ -135,6 +161,21 @@ module.exports = {
         }
       });
     }
+    if (this.options.componentAuthoringFormat === 'strict') {
+      const strictFilesToRemove =
+        this.options.isTypeScriptProject || this.options.typescript ? '.gjs' : '.gts';
```

Comment:
I'm not 100% sure if the `isTypeScriptProject` check is needed. It's possible that when that is true, `options.typescript` is also true, but I would have to double check.